silence/silence.go

```diff
@@ -327,6 +327,9 @@ func New(o Options) (*Silences, error) {
 			}
 		} else {
 			o.SnapshotReader = r
+			defer func(fileDescriptor *os.File) {
```

Comment:
I think this could be just `defer r.Close()` or does that fail linting because the error is not checked?

Reply:
It could, yes. I think this is more a question of style. I think this is more explicit.